lib/traject/macros/custom.rb

```diff
@@ -205,12 +205,22 @@ def self.includes_oclc_indicators?(sf_a)
           sf_a.include?('OCLC')
       end
 
-      # Extract ht_id
-      def extract_ht_id
+      def extract_hathi_data
         lambda do |_record, accumulator, context|
           oclc_number = context.output_hash&.dig('oclc_number_ssim')&.first
-          accumulator << HATHI_ETAS_OVERLAP[oclc_number]
-          accumulator.compact
+          ht_hash = HATHI_MULTI_OVERLAP&.dig(oclc_number)&.first || HATHI_MONO_OVERLAP&.dig(oclc_number)&.first
+          accumulator << ht_hash.to_json unless ht_hash.nil?
+        end
+      end
+
+      def hathi_to_hash(ht_format)
+        hathi_overlap_csv = "#{settings['hathi_overlap_path']}#{settings["hathi_#{ht_format}_overlap_file"]}"
+        CSV.read(hathi_overlap_csv)
+           .group_by(&:shift)
+           .each do |_oclc, ht_data|
+          ht_key = ht_format == 'mono' ? :ht_id : :ht_bib_key
+          ht_data.map! { |data| [ht_key, :access].zip(data).to_h }
+          ht_data.reject! { |data| data[:access] == 'deny' } if ht_data.length > 1
```

Comment:
I'm curious, is `ht_data.length` ever _more_ than 2? Not asking for a change to be clear, just curious. ht_data should be something like `{mono: 10, access: "deny"}`. And for the cases where there's only `1` that's because `access` is `nil` right?

Reply:
the final filtered files  produced from the rake task should only have one of each allow or deny, because we are doing uniq on them.

Reply:
ok. cool. I should have just checked there. Thanks!